tempodb/pool/pool.go

```diff
@@ -29,11 +29,10 @@ var (
 	})
 )
 
-type JobFunc func(ctx context.Context, payload interface{}) ([]byte, string, error)
+type JobFunc func(ctx context.Context, payload interface{}) (interface{}, error)
```

Comment:
Not something to address now, but thoughts on killing the job pool in favor of just using the boundedwaitgroup and goroutines?

The only reason I can think to keep this code long term is b/c it preserves order of jobs. So if multiple queries are hitting at the same time this should reduce long tail latency by completely finishing one trace by id query before moving to the next.

Reply:
Yes had the same thoughts exactly (while editing this I had fond memories of tuning pool size for large block lists).  But leaned toward making that change in a separate PR.